saleor/shipping/models.py

```diff
@@ -209,6 +227,41 @@ def get_total(self):
         return self.price
 
 
+class ShippingMethodChannelListing(models.Model):
+    shipping_method = models.ForeignKey(
+        ShippingMethod,
+        null=False,
+        blank=False,
+        related_name="channel_listing",
+        on_delete=models.CASCADE,
+    )
+    channel = models.ForeignKey(
+        Channel,
+        null=False,
+        blank=False,
+        related_name="shipping_method_listing",
+        on_delete=models.CASCADE,
+    )
+    price = models.DecimalField(
```

Comment:
Why `price`, `min_value` and `max_value` are still on `ShippingMethod` model?

Reply:
Instead of `price` as the decimal field, you should move `price` as MoneyField with require `currency`  and `price_amount`.

Reply:
Same situation with `minimum_order_price` and `maximum_order_price` should be `MoneyField`